File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        int c=0;
        StringBuilder sb=new StringBuilder();
        for(char t:s.toCharArray()){
            if(t=='('){
                if(c>0){
                    sb.append(t);
                }
                c++;
            }else{
                if(c>1){
                    sb.append(t);
                }
                c--;
            }
        }
        return sb.toString();
    }
}

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        res=[]
        c=0
        for i in s:
            if i=='(':
                if c>0:
                    res.append(i)
                c+=1
            else:
                if c>1:
                    res.append(i)
                c-=1
        return "".join(res)